arch/arm/CMakeLists.txt

```diff
@@ -17,5 +17,10 @@ zephyr_compile_options(
   ${ARCH_FLAG}
   )
 
+zephyr_compile_options_ifdef(CONFIG_COVERAGE
```

Comment:
Any reason why we are doing this per architecture, can't we enable this globally?

Reply:
Globally enabling this will make gcov instrument test code aswell. usage related issue with Cmake. And it is simpler to make it instrument only the code that is actually needed.